### ISSUE TITLE
Ensure Deps

### ISSUE THREAD

**Issue by AurevoirXavier** [darwinia-network/darwinia-common#1]
```sh
# HARD FAILING
MUST_NOT=(
	"client crates must not depend on anything in /frame"
	"client crates must not depend on anything in /node"
	"frame crates must not depend on anything in /node"
	"frame crates must not depend on anything in /client"
	"primitives crates must not depend on anything in /frame"
)

# ONLY DISPLAYED, script still succeeds
PLEASE_DONT=(
	"primitives crates should not depend on anything in /client"
)

VIOLATIONS=()
PACKAGES=()
```

**Comment by yanganto** [comment]
These rules may be written in pre-commit script `.maintain/hooks/pre-commit`, such that we don't need to check again.

**Comment by AurevoirXavier** [comment]
@yanganto https://github.com/paritytech/substrate/blob/master/.maintain/ensure-deps.sh

**Comment by yanganto** [comment]
Great.  The pre-commit script should call `ensure-deps.sh`, and I will make a PR on this.

**Comment by yanganto** [comment]
The ensure dependency script is merged in #13

**Comment by HackFisher** [comment]
duplicate with #23
